### ISSUE TITLE
`eb local run` supports docker-compose.yml?

### ISSUE THREAD

**Issue by acomagu** [aws/aws-elastic-beanstalk-cli#58]
The [Beanstalk Documentation](https://docs.aws.amazon.com/elasticbeanstalk/latest/dg/single-container-docker-configuration.html) says:

> Create a docker-compose.yml file to deploy a Docker image from a hosted repository to Elastic Beanstalk. No other files are required if all your deployments are sourced from images in public repositories.

But if execute `eb local run` in the directory includes only docker-compose.yml, it fails with a message:

> ERROR: NotFoundError - The EB CLI cannot find Dockerfile or the Dockerrun.aws.js

Doesn't `eb local run` support docker-compose? Any plan to do?

**Comment by Palakpatel67** [comment]
@acomagu thanks for reporting this issue. Currently`eb local run` does not support `docker-compose.yml`. We do have plans to support this. thanks!

**Comment by nikitamendelbaum** [comment]
> @acomagu thanks for reporting this issue. Currently`eb local run` does not support `docker-compose.yml`. We do have plans to support this. thanks!

if so - there is a mistake in https://docs.aws.amazon.com/elasticbeanstalk/latest/dg/docker.html#single-container-docker.test-remote where it clearly says to use that command with docker-compose.yml pre-created.

**Comment by vitsensei** [comment]
Any update for this? I would benefit greatly if this is possible.

**Comment by tsoisauce** [comment]
Yes, this would be very helpful, especially with troubleshooting. I am experiencing issues deploying with `docker-compose.yml`

**Comment by tarasBilobran** [comment]
any updates on this one?

**Comment by jonaseriksson84** [comment]
Any news?

**Comment by anonymousmaharaj** [comment]
Still need this.

**Comment by BugliL** [comment]
Same here

**Comment by cam-biolexis** [comment]
Or at least update your docs. I spent forever thinking this was possible and then just found this

**Comment by cidraljunior** [comment]
Any news?

**Comment by dan1229** [comment]
Any news on this? Spent about a week playing with this to find this thread.

**Comment by ThierDev** [comment]
Still needed 👍 
 the deployment of docker-compose.yml is simply not working online as well  :/ says cannot find Dockerfile or the Dockerrun.aws.json file while in fact, the docker-compose.yml file is here.

**Comment by rahmataw** [comment]
> Still needed 👍 the deployment of docker-compose.yml is simply not working online as well :/ says cannot find Dockerfile or the Dockerrun.aws.json file while in fact, the docker-compose.yml file is here.

yup i have the same issue. It is like the EB not read docker-compose.yml when I upload as zipped file. But it can run when I only upload docker-compose.yml.

**Comment by VaZark** [comment]
it'd nice to see support _docker-compose_ added so that I can replicate my env perfectly on my local machine. :+1:

**Comment by baselka** [comment]
docker-compose support locally is indeed needed 😃 👏

**Comment by samucancld** [comment]
we need it asap

**Comment by chiragpurohit71085** [comment]
seems there is no update on this

**Comment by nickbreen** [comment]
The documentation literally states this is supported: https://docs.aws.amazon.com/elasticbeanstalk/latest/dg/single-container-docker-configuration.html

But obviously it isn't.

Please, at least, correct the docs.

**Comment by avimehenwal** [comment]
After 3 years, the docs are still misleading and compose still doesnt work with `eb local`.
